test/integration/puppetlabs/services/master/environment_modules_int_test.clj

```diff
@@ -0,0 +1,78 @@
+(ns puppetlabs.services.master.environment-modules-int-test
+  (:require [clojure.test :refer :all]
+            [puppetlabs.http.client.sync :as http-client]
+            [puppetlabs.kitchensink.core :as ks]
+            [puppetlabs.puppetserver.bootstrap-testutils :as bootstrap]
+            [puppetlabs.trapperkeeper.testutils.logging :as logutils]
+            [puppetlabs.puppetserver.testutils :as testutils]
+            [cheshire.core :as cheshire]
+            [me.raynes.fs :as fs]
+            [clojure.tools.logging :as log]
+            [puppetlabs.services.jruby.jruby-puppet-testutils :as jruby-puppet-testutils])
+  (:import (com.puppetlabs.puppetserver JRubyPuppetResponse JRubyPuppet)
+           (java.util HashMap)))
+
+(def test-resources-dir
+  "./dev-resources/puppetlabs/services/master/environment_modules_int_test")
+
+(use-fixtures :once
+              (testutils/with-puppet-conf
+               (fs/file test-resources-dir "puppet.conf")))
+
+(defn get-env-modules
+  ([env-name]
+   (get-env-modules env-name nil))
+  ([env-name if-none-match]
+   (let [opts (if if-none-match
+                {:headers {"If-None-Match" if-none-match}})]
+     (try
+       (http-client/get
+        (str "https://localhost:8140/puppet/v3/"
+             "environment_modules?"
+             "environment="
+             env-name)
+        (merge
+         testutils/ssl-request-options
+         {:as :text}
+         opts))
+       (catch Exception e
+         (throw (Exception. "environment_modules http get failed" e)))))))
+
+(defn response->module-info-map
+  [response]
+  (-> response :body cheshire/parse-string))
+
+(deftest ^:integration environment-modules-integration-cache-disabled-test
+  (testing "when environment modules cache is disabled for a module request"
+    (bootstrap/with-puppetserver-running-with-config
+     app
+     (-> {:jruby-puppet {:max-active-instances 1}}
+         (bootstrap/load-dev-config-with-overrides)
+         (ks/dissoc-in [:jruby-puppet
+                        :environment-class-cache-enabled]))
+     (logutils/with-test-logging
+       (let [foo-file (testutils/write-foo-pp-file
```

Comment:
Ah, but I guess the metadata.json file is written behind this call so nevermind.